tests/api_connexion/endpoints/test_dag_run_endpoint.py

```diff
@@ -1154,3 +1154,91 @@ def test_should_raises_403_unauthorized(self):
             environ_overrides={'REMOTE_USER': "test_view_dags"},
         )
         assert response.status_code == 403
+
+
+class TestPostSetDagRunState(TestDagRunEndpoint):
+    @parameterized.expand([("failed",), ("success",)])
+    @pytest.fixture(scope="module")
+    def test_should_respond_200(self, state, dag_maker):
+        dag_id = "TEST_DAG_ID"
+        with create_session() as session:
+            dag = DagModel(dag_id=dag_id)
+            dag_run = dag_maker.create_dagrun()
+            session.add(dag)
+            session.add(dag_run)
```

Comment:
```suggestion
    def test_should_respond_200(self, state, dag_maker):
        dag_id = "TEST_DAG_ID"
        dag_run_id = 'TEST_DAG_RUN_ID'
        with dag_maker(dag_id):
            DummyOperator(task_id='task_id')
        dag_maker.create_dagrun(run_id=dag_run_id)
```

Reply:
@ephraimbuddy I can't get this setup to work and don't really understand it, I guess, so a couple of questions:
1. How can I inject dag_maker without using `@pytest.fixture`, if I'm also using `@parameterized.expand`? I get an error that I'm missing an error `missing 1 required positional argument: 'dag_maker'`
2. What is the purpose of DummyOperator here, and why run in separately in a context manager from `dag_maker.create_dagrun()`?
I'm considering reverting the usage of this and just creating the DAG using the constructor, like the other tests in this file...

Reply:
> How can I inject dag_maker […]?

Just include `dag_maker` as an argument. Pytest magically picks the argument up and injects the fixture into the function. But to make the test compatible with pytest, you need to use `pytest.mark.parametrize` instead of `parameterized.expand`.

So something like

```python
@pytest.mark.parametrize("state", [...])
def test_should_respond_200(self, dag_maker, state):
    ...
```

> What is the purpose of DummyOperator here, and why run in separately in a context manager from `dag_maker.create_dagrun()`?

The DummyOperator will be included as a part of the DAG, similar to

```python
with DAG(...):
    MyOperator(...)
```

It’s not actually used anywhere in the test, so technically you can just do

```python
with dag_maker(...):
    pass
```

A DAG without any operators is kind of weird, but allowed.

Reply:
Thanks for the explanation! Makes sense now.